src/transformers/training_args.py

```diff
@@ -493,6 +499,13 @@ def _setup_devices(self) -> "torch.device":
         elif is_torch_tpu_available():
             device = xm.xla_device()
             self._n_gpu = 0
+        elif is_sagemaker_distributed_available():
+            import smdistributed.dataparallel.torch.distributed as dist
+
+            dist.init_process_group()
```

Comment:
Shouldn't we then use `dist` alias consistently everywhere then? it'd be easier to read the code 

There is a whole bunch of different variations now - the code in this branch is identical to the 2 branches down, other than the import.

Reply:
or alternatively we replicate this branch for every integration - one way or another but so it'd be consistent.

Reply:
It's harder to have clean imports in this file since it needs to have a hard-dependency on no framework (so we can't just do the import at the top). Maybe I should just have done `import smdistributed` here and then use the full path. For the other files, the `dist` is used consistently.

Reply:
```
if is_sagemaker_distributed_available():
            import smdistributed.dataparallel.torch.distributed as dist
else:
            import torch.distributed as dist
[...]
dist.nit_process_group()
```

Reply:
oh, and I see you've already done this exact thing in `trainer_pt_utils.py` in this PR.

Reply:
Yes but this can't be done here since the else would need to be `elif is_torch_available()` and then we take the risk of having `dist` not defined somehow. Plus the `init_process_group` command needs to be invoked with different arguments in both cases, so we can't regroup them.

Reply:
Indeed,  `init_process_group` w/o an arg. I missed that.

What I'm leaning towards then is having a full copy of all these settings for each integration branch, i.e. undoing the deepspeed merge with the `else` fallback and giving it a full branch, like you did for sage, so that each integration is self contained and easier to take in, with the cost of small amount of duplication.

What do you think, @sgugger?

Reply:
That's a great idea!

Reply:
Thank you for your validation, @sgugger - I will then recode it at some point unless someone beats me to it.